## Introduction
- Name: Izzan Dienurrahman
- Batch: HCK-003

## Objective
- Develop a forecasting model based on nutritional information of plants that has been gathered from a laboratory experiment.

## Problem Statements
- The current understanding of the nutritional value of plants is insufficient.
- Laboratory experiments provide valuable data on plant nutrition.
- There is a need to develop a prediction model based on the nutritional information gathered from these experiments.
- The model must be accurate and reliable in forecasting the nutritional content of plants.
- Improving our knowledge of plant nutrition will have a positive impact on plant health and yield.


## Dataset Description
`aria_data.xlsx`
- **target** contained information to predict plant nutrition
- **V1 - V8** contained information about variable to predict plant nutrition
- **sample_type** contained information about 2 different labs sample obtained

## Library

In [1]:
# import all libraries needed for this project
import pandas as pd

## Data Loading

In [3]:
# read files into pandas dataframe
df = pd.read_excel('aria_data.xlsx')

In [4]:
# show top 5 data entries
df.head()

target          v1          v2          v3          v4          v5  \
0    4.63  260.600000  199.000000  391.333333  371.733333  466.800000   
1    5.00  263.200000  194.600000  393.666667  348.533333  401.800000   
2    4.91  247.266667  205.800000  375.466667  354.866667  419.000000   
3    4.77  257.944444  191.833333  368.722222  334.666667  417.222222   
4    4.89  258.200000  195.733333  356.666667  333.333333  394.600000   

           v6          v7           v8 sample_type  
0  273.133333  742.066667  4047.733333       lab 1  
1  220.066667  712.000000  4673.466667       lab 1  
2  222.133333  677.200000  4386.800000       lab 1  
3  195.944444  678.555556  4588.777778       lab 1  
4  196.000000  677.133333  4867.066667       lab 1

In [5]:
# show bottom 5 data entries
df.tail()

target        v1          v2          v3          v4       v5  \
155    4.83  573.2500  361.250000  643.125000  489.812500  666.875   
156    4.47  678.3750  398.125000  678.250000  541.125000  704.000   
157    4.63  634.0000  382.866667  657.466667  505.066667  719.200   
158    4.35  615.3125  366.750000  642.937500  483.687500  675.625   
159    4.43  575.2000  350.266667  648.333333  495.400000  679.800   

             v6          v7           v8 sample_type  
155  355.250000  712.312500  3943.687500       lab 2  
156  393.500000  788.812500  3928.187500       lab 2  
157  366.600000  792.133333  3861.200000       lab 2  
158  359.062500  730.125000  3859.062500       lab 2  
159  346.266667  718.733333  3903.333333       lab 2

In [6]:
# show brief info
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 160 entries, 0 to 159
Data columns (total 10 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   target       160 non-null    float64
 1   v1           160 non-null    float64
 2   v2           160 non-null    float64
 3   v3           160 non-null    float64
 4   v4           160 non-null    float64
 5   v5           160 non-null    float64
 6   v6           160 non-null    float64
 7   v7           160 non-null    float64
 8   v8           160 non-null    float64
 9   sample_type  160 non-null    object 
dtypes: float64(9), object(1)
memory usage: 12.6+ KB


There are **160** entries of data with **10** columns and there are no missing values. All data are of the numerical type except `sample_type`.

In [8]:
# show duplicated data
df.duplicated().sum()

0

There is no duplicated data either.

## Exploratory Data Analysis (EDA)

In [7]:
# describe data with common stats such as count, min, max, etc.
df.describe()

target          v1          v2          v3          v4          v5  \
count  160.00000  160.000000  160.000000  160.000000  160.000000  160.000000   
mean     4.73375  389.892437  237.442677  480.573041  394.109622  487.316520   
std      0.21405  148.673511   70.128690  118.798984   63.060442  100.941711   
min      4.13000  227.285714  178.800000  348.933333  313.733333  373.333333   
25%      4.60000  268.608333  194.029018  384.419643  353.181250  418.583333   
50%      4.72000  295.162500  204.187500  414.383929  370.741667  456.031250   
75%      4.89000  536.409375  244.716667  591.611404  424.131250  496.475000   
max      5.26000  678.375000  422.812500  722.312500  558.500000  721.000000   

               v6          v7           v8  
count  160.000000  160.000000   160.000000  
mean   251.450820  714.737926  4456.913233  
std     59.634652   51.148098   335.035933  
min    189.200000  586.266667  3725.666667  
25%    213.000000  678.385417  4273.437500  
50%    226.062500  718.833333  4554.766667  
75%    262.531250  751.446875  4700.612500  
max    415.375000  853.466667  5086.375000

## Data Preprocessing

## Feature Selection

## Feature Engineering

## Model Definition

## Model Evaluation

## Conclusion

## Further Improvement